src/screens/search_results/SearchResultsHeader.tsx

```diff
@@ -109,100 +126,109 @@ const SearchResultsHeader: React.FC<SearchResultsHeaderProps> = ({
 
     const handleViewToggle = (view: 'grid' | 'list') => {
         setViewState?.(view);
-        localStorage.setItem('streamabilityView', view);
+        localStorage.setItem(viewStateKey, view);
         setHash?.(Math.random());
     };
 
     return (
         <div className='flex flex-col md:flex-row flex-wrap justify-between align-middle w-full p-3'>
             <Typ variant='h5' alignSelf='center' margin={1}>
-                Search results for: <span className='underline'>{query}</span>
+                {query && (
+                    <p>
+                        Search results for: <span className='underline'>{query}</span>
+                    </p>
```

Comment:
We should use `Typography` element instead of `p`. 
We should also only prepend the "Search results for: " on the search results screen, not the discover detail.